obelisk/cpp/hardware/robots/unitree/g1_interface.h

```diff
@@ -50,40 +46,42 @@ namespace obelisk {
                 }
             }
 
-            // Setup local variables
-            // TODO: For now we are not supporting the hands here
+            // Setup local variables (these will have UnitreeInterface ordering)
             joint_pos_.resize(G1_27DOF + G1_EXTRA_WAIST);
             joint_vel_.resize(G1_27DOF + G1_EXTRA_WAIST);
             start_user_pose_.resize(G1_27DOF + G1_EXTRA_WAIST);
             user_pose_.resize(G1_27DOF + G1_EXTRA_WAIST);
 
+            // Get user pose
             this->declare_parameter<std::vector<double>>("user_pose", user_pose_);
             user_pose_ = this->get_parameter("user_pose").as_double_array();
+
+            // Handle default joint names
+            std::vector<std::string> default_names_vec(G1_27DOF_JOINT_NAMES.begin(), G1_27DOF_JOINT_NAMES.end());
+            this->declare_parameter<std::vector<std::string>>("default_joint_names", default_names_vec);
+            default_joint_names_ = this->get_parameter("default_joint_names").as_string_array();
+            default_joint_mapping_.clear();
+            for (size_t i = 0; i < default_joint_names_.size(); ++i) {
+                default_joint_mapping_.emplace(default_joint_names_[i], static_cast<int>(i));
```

Comment:
Maybe I'm missing it - but is there any place here where we are checking that the default names and valid and erroring if not?